2-projects/1-todo-list/script.js

```diff
@@ -17,9 +55,23 @@ function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
   // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  let todoInputEl = document.querySelector('#todoInput').value; // Reads the value of the #todoInput element
+  let todoObj = { task: todoInputEl, completed: false }; // Assigns a new task to an object
+  todos.push(todoObj); // Pushes the todoObj to global todos array
+  let newTodo = [todoObj]; // Adds todoObj to an array
+  populateTodoList(newTodo); // This function is called now with the newTodo as the parameter
+  document.querySelector('#todoInput').value = ''; // Resets the form input field to an empty field
 }
 
-// Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
-function deleteAllCompletedTodos() {
-  // Write your code here...
+// Advanced challenge: Write a function that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
+function deleteAllCompletedTodos(event) {
+  event.preventDefault();
+  let list = document.getElementById('todo-list');
+  let todoEL = document.querySelectorAll('li');
+  todoEL.forEach((todoLi) => {
+    // Deletes the todo text which only have a line through style
+    if (todoLi.style.textDecoration === 'line-through') {
```

Comment:
this is another example of the logic being dependent on the style.

It is harder in this case because you do not have access to the `todo.completed` boolean. In this case it would be best to add and remove a `.completed` class to each todo inside your event listener.

And then deleteAllCompletedTodos  is as simple as `document.querySelectorAll('.completed');` and deleting them, no logic needed